presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConfig.java

```diff
@@ -451,6 +455,19 @@ public HiveConfig setImmutablePartitions(boolean immutablePartitions)
         return this;
     }
 
+    public InsertExistingPartitionsBehavior getInsertExistingPartitionsBehavior()
+    {
+        return insertExistingPartitionsBehavior;
+    }
+
+    @Config("hive.insert-existing-partitions-behavior")
+    @ConfigDescription("Default value for insert existing partitions behavior")
+    public HiveConfig setInsertExistingPartitionsBehavior(InsertExistingPartitionsBehavior insertExistingPartitionsBehavior)
+    {
+        this.insertExistingPartitionsBehavior = insertExistingPartitionsBehavior;
```

Comment:
```suggestion
        this.insertExistingPartitionsBehavior = requireNonNull(insertExistingPartitionsBehavior, "insertExistingPartitionsBehavior is null");
```

Reply:
I don't see how this enum value could be set to null. 
In my opinion it is set by airlift and in case there no such property in config file method never will be invoked, otherwise only correct enum value will be assigned.

Reply:
Usually we don't see how `null` are passed until the moment when we get NPE. It is yet another defensive programming. airlift can have bugs, we want to catch them as soon as possible.

Reply:
Nevertheless I don't think it worth to check NPE here, but I've applied your suggestion.